main.c

```diff
@@ -74,12 +79,129 @@ void usage(int error)
 	fprintf(dest, "        -v               : Verbose\n");
 	fprintf(dest, "        -s               : Signed using bootsig.bin\n");
 	fprintf(dest, "        -0/1/2/3/4/5/6   : Only look for CMs attached to USB port number 0-6\n");
-	fprintf(dest, "        -p [pathname]    : Only look for CM with USB pathname\n");
+	fprintf(dest, "        -p [pathname]    : Only look for CM with USB pathname\n");\
+	fprintf(dest, "        -i [serialno]    : Only look for a Raspberry Pi Device with a given serialno\n");
 	fprintf(dest, "        -h               : This help\n");
 
 	exit(error ? -1 : 0);
 }
 
+libusb_device_handle * LIBUSB_CALL open_device_with_serialno(
+	libusb_context *ctx, char* serialno
+) {
+	struct libusb_device **devices;
+	struct libusb_device *cursor;
+	struct libusb_device_handle *handle = NULL;
+	int r = 0;
+
+	if (libusb_get_device_list(ctx, &devices) < 0)
+		return NULL;
+
+	uint32_t device_index = 0;
+	unsigned char *serial_buffer = calloc(33, sizeof(uint8_t));
+	if (serial_buffer == NULL) {
```

Comment:
Drop the braces on one-line if bodies like this.

Reply:
Understood if that's the style here - I habitually use braces even on single-line bodies after the Apple `goto fail` kerfuffle.

Reply:
The existing code uses braces inconsistently, but no-braces seems to win over braces for this case. Plus it's the kernel coding style and therefore irreproachable.